frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
   SearchRoot,
 } from "./SearchApp.styled";
 
-const PAGE_SIZE = 50;
-
-const SEARCH_FILTERS = [
-  {
-    name: t`Apps`,
-    filter: "app",
-    icon: "star",
-  },
-  {
-    name: t`Dashboards`,
-    filter: "dashboard",
-    icon: "dashboard",
-  },
-  {
-    name: t`Collections`,
-    filter: "collection",
-    icon: "folder",
-  },
-  {
-    name: t`Databases`,
-    filter: "database",
-    icon: "database",
-  },
-  {
-    name: t`Models`,
-    filter: "dataset",
-    icon: "model",
-  },
-  {
-    name: t`Raw Tables`,
-    filter: "table",
-    icon: "table",
-  },
-  {
-    name: t`Questions`,
-    filter: "card",
-    icon: "bar",
-  },
-  {
-    name: t`Pulses`,
-    filter: "pulse",
-    icon: "pulse",
-  },
-  {
-    name: t`Metrics`,
-    filter: "metric",
-    icon: "sum",
-  },
-  {
-    name: t`Segments`,
-    filter: "segment",
-    icon: "segment",
-  },
-];
-
 export default function SearchApp({ location }) {
   const { handleNextPage, handlePreviousPage, setPage, page } = usePagination();
-  const [filter, setFilter] = useState(location.query.type);
 
-  const handleFilterChange = filterItem => {
-    setFilter(filterItem && filterItem.filter);
-    setPage(0);
-  };
+  const searchText = useMemo(
+    () => getSearchTextFromLocation(location),
+    [location],
+  );
 
-  const query = {
-    q: location.query.q,
-    limit: PAGE_SIZE,
-    offset: PAGE_SIZE * page,
-  };
+  const searchFilters = useMemo(() => {
+    return getFiltersFromLocation(location);
+  }, [location]);
+
+  const [selectedSidebarType, setSelectedSidebarType] = useState(null);
+
+  useEffect(() => {
+    setSelectedSidebarType(null);
+  }, [searchText, searchFilters]);
+
+  const query = useMemo(
+    () => ({
+      q: searchText,
+      ..._.omit(searchFilters, SearchFilterKeys.Type),
+      models: selectedSidebarType ?? searchFilters[SearchFilterKeys.Type],
```

Comment:
What's the rationale behind using `searchFilters[SearchFilterKeys.Type]`? Can't we just use `searchFilters.type`? It feels too complicated to me.

Reply:
It's something I've been struggling with. Personally, I prefer to use keys like this rather than relying on the IDE+typescript for figuring out which keys are in an object, and for me seems more readable (just me though). Happy to be pushed back on this though

Reply:
I think, in this case, it also looks better than 
```
..._.omit(searchFilters, "type"),
models: selectedSidebarType ?? searchFilters.type (or even searchFilters['type']),
```

Reply:
one of the things tha's complicated is that `SearchFilterKeys` isn't just a type it's defined as `const` and I don't think it should be in `types.ts`. In the new future, we're gonna force `import type` for types and this won't work as `import type`. As `SearchFilterKeys.Type` is used as a value here.

But coming back to your example, I can understand the motivation. I just don't think that's consistent with the rest of the code base. If you can find examples where we use this pattern I'd be happy to 👍 this.

Reply:
Yeah, I couldn't get away from setting `SearchFilterKeys` as a type because we use those values to get data from the URL here, and I didn't want to have duplicate type/value pairs to maintain

https://github.com/metabase/metabase/blob/331993c362699904adeae00b47d303f5f307d71d/frontend/src/metabase/search/utils/search-location/search-location.ts#L23-L30

Reply:
I you can hold on, I want to try to come up with something, but I'm about to EoD. I still think the types in `frontend/src/metabase/search/types.ts` are too complicated and can be simplified further.

Reply:
I found the example I was thinking of here in `QueryModals`:

https://github.com/metabase/metabase/blob/0f897c1d28e89445471c21a4d2fba85018b4b558/frontend/src/metabase/query_builder/constants.ts#L1-L8

https://github.com/metabase/metabase/blob/0f897c1d28e89445471c21a4d2fba85018b4b558/frontend/src/metabase/query_builder/components/QueryModals.tsx#L52


and another one in `ValidationError`:

https://github.com/metabase/metabase/blob/0f897c1d28e89445471c21a4d2fba85018b4b558/frontend/src/metabase-lib/ValidationError/constants.ts#L1-L3

https://github.com/metabase/metabase/blob/0f897c1d28e89445471c21a4d2fba85018b4b558/frontend/src/metabase-lib/ValidationError/types.ts#L1-L4

I can move the `SearchFilterKeys` object into a `constants.ts` file (it doesn't belong in `types` as you said) to keep the pattern consistent with these

Reply:
Sounds good. I think moving values to `constants.ts` and using some of that in the types sounds better and looks more consistent with the codebase.